### PR TITLE
[AR Number] Coverage Improvements and Spec Updates

### PR DESCRIPTION
**Pull request by neudurgeshp**
**PR Includes**
- Number Definitions
- Specs
   - NumberModel
- Number Model Extractor
   - Total Cases 279: (Extractor Passed - 125, Full Passed - 37, Skipped - 154)
   - New Cases 135: (Extractor Passed - 117, Full Passed - 29, Skipped - 18)
   - Integer 48: Extraction Passed - 42, Full Passed - 6, Skipped - 6
   - Fraction 56: Extraction Passed - 47, Full Passed - 12, Skipped - 9
   - Decimal 9: Extraction Passed - 9
   - Power 11: Extraction Passed - 8, Full Passed - 0, Skipped - 3
   - Negative 11: Extraction Passed - 11, Full Passed - 11

**Some of the cases from earlier iteration that we added are modified. We identified some discrepancy in those cases and got re-corrected by Language Expert. Below are inputs that are modified.
- تسعين وخمس مائة أخماس من الوزن
- كان النتيجة مئة أجزاء من الف وخمسة
- واحد من وخمسة وعشرون اجزاء المقطع تم تنزيله
- سيمديني ان اعطيك  ٤ آلاف و ٣ مائة و٢١ ريال.
- سالب مائة وثلاثون أخماس

One duplicate case removed.
- واحد على ثلاثة

### CODE REVIEW

File: Specs/Number/Arabic/NumberModel.json [hunk 1]
```diff
@@ -1417,19 +1407,19 @@
     ]
   },
   {
-    "Input": "تسعين وخمس مائة أخماس من الوزن",
+    "Input": "كم هي خمسة وتسعون مائة على خمسة ؟",
```
Comment (tellarin) [inline Specs/Number/Arabic/NumberModel.json:1410]: Can you provide the reasons for the spec edits and removals in the PR description please?
Keep in mind that we want to keep as many variations as possible. Spec cases should only be removed if they really don't apply in a target language. The original writer of these specs was a native Arabic speaker from Egypt, so if these are not MSA, they may still be of frequent use.

File: Specs/Number/Arabic/NumberModel.json [hunk 2]
```diff
@@ -2001,9 +1980,9 @@
     ]
   },
   {
-    "Input": "عدد الطلاب في الدولة ١،٢٣٤،٥٦٧",
+    "Input": "الأرقام هي ١ ، ٢٣٤ ، ٥٦٧",
```
Comment (tellarin) [inline Specs/Number/Arabic/NumberModel.json:1983]: Same, and so on.

File: Specs/Number/Arabic/NumberModel.json (hunk 1 above)
Comment (neudurgeshp) [reply Specs/Number/Arabic/NumberModel.json:1410]: Sure @tellarin , I just added in PR Description.
FYI,  we identified some discrepancy in some cases and got re-corrected by Language Expert. 
Thanks

File: Specs/Number/Arabic/NumberModel.json (hunk 2 above)
Comment (neudurgeshp) [reply Specs/Number/Arabic/NumberModel.json:1983]: Added brief in PR Description.

Comment (tellarin) [reply Specs/Number/Arabic/NumberModel.json:1983]: I'll merge, but can you add more info there about the types of issues that led to the corrections? Wrong grammar, not exactly the same meaning, invalid sentences? This is very valuable feedback to track. Thanks!

**Pull request merged**